package/MDAnalysis/lib/pkdtree.py

```diff
@@ -0,0 +1,170 @@
+        """
+        wrapped_c = apply_PBC(center_point.reshape(1, 3), self.box)[0]
+        extents = self.box/2.0
+        extents = np.where(extents > radius, radius, extents)
+        # displacements are vectors that we add to wrapped_c to
+        # generate images "up" or "down" the central cell along
+        # the axis that we happen to be looking.
+        #
+        # TO-DO: extend to a triclinic box by finding distance from wrapped_c
+        # to each of the planes enclosing the central cell.
+        displacements = list()
+        for i in range(self.dim):
+            displacement = np.zeros(self.dim)
+            extent = extents[i]
+            if extent > 0.0:
+                if self.box[i] - wrapped_c[i] < extent:
+                    displacement[i] = -self.box[i]  # "lower" image
+                    displacements.append(displacement)
+                elif wrapped_c[i] < extent:
+                    displacement[i] = self.box[i]  # "upper" image
+                    displacements.append(displacement)
+        # If we have displacements along more than one axis, we have
+        # to combine them. This happens when wrapped_center is close
+        # to any edge or vertex of the central cell.
+        # face, n_displacements==1; no combination
+        # edge, n_displacements==2; combinations produce one extra displacement
+        # vertex, n_displacements==3; five extra displacements
+        n_displacements = len(displacements)
+        if n_displacements > 1:
+            for start in range(n_displacements - 1, -1, -1):
+                for i in range(start+1, len(displacements)):
+                    displacements.append(displacements[start]+displacements[i])
+        return [wrapped_c, ] + [wrapped_c + d for d in displacements]
+
+    def search(self, center, radius):
+        """Search all points within radius of center and its periodic images.
+        Wrapping of center coordinates is enforced to enable comparison to
+        wrapped coordinates of points in the tree.
+
+        Parameter
+        ---------
+        center: NumPy.array
+          origin around which to search for neighbors
+        radius: float
+          maximum distance around which to search for neighbors. The search
+          radius is half the smallest periodicity if radius exceeds this value
+        """
+        if not self.built:
+            raise RuntimeError('Unbuilt tree. Run tree.set_coords first')
+        if center.shape != (self.dim,):
+            raise ValueError('Expected a ({},) NumPy array'.format(self.dim))
+        self._indices = set()  # clear previous search
+        for c in self.find_centers(center, radius):
+            self.kdt.search_center_radius(c, radius)
+            new_indices = self.kdt.get_indices()  # returns None or np.array
+            if new_indices is not None:
+                self._indices.update(new_indices)
+        self._indices = sorted(list(self._indices))
```

Comment:
In the end I initialized the seach with a `set` because duplicates are discarded when updating such object. I also realized we don't need a numpy array for the final list of indices, a python list works equally well when passed onto `AtomNeighborSearch.search`